### PR TITLE
check for Windows version

### PR DESCRIPTION
**Pull request by rfomin**
direct3d11 broke screenshots on my system

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
You should probably actually use this check. 😉

File: win32/win_version.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#define WIN32_LEAN_AND_MEAN
+
+#include "win_version.h"
+
+#include <windows.h>
+#include <stdlib.h>
+
+typedef long (__stdcall *PRTLGETVERSION)(PRTL_OSVERSIONINFOEXW);
+
+int I_CheckWindows11(void)
```
[the remark this answers is not in the document]
Comment (rfomin) [reply win32/win_version.c:28]: It's not that simple, MS changed the versioning scheme several times. This function checks only Windows 11.

**Comment by rfomin** [comment]
> Nope, screnshots are garbage in Win 11 as of master HEAD.

I think we should abandon the direct3d11 approach for now. Or write an alternate screenshot function.

**Comment by rfomin** [comment]
> Yes, let's just ditch it for now. How about @JNechaevsky's approach?

Ok if it works. I can't test it.

**Comment by JNechaevsky** [comment]
Oh my, my comment again. 🤤

I have a strong suspicion that it's Windows-only bug. AFAIR, vsync toggling code written by Fabian still working perfect on WinXP, even with latest SDL libraries (or was it 2.0.18?), so probably something changed in Direct3D API on Microsoft side, but what and how - really no idea.

**Comment by rfomin** [comment]
> so probably something changed in Direct3D API on Microsoft side, but what and how - really no idea.

From my understading there is a bug in Windows 11 Direct3D9 driver.

It seems `SDL_RenderReadPixels`  will be fixed in SDL 2.0.22: https://github.com/libsdl-org/SDL/issues/4782

**Comment by JNechaevsky** [comment]
Good to know, thank you! Looks likes 2.0.22 is already underway: https://www.libsdl.org/tmp/download-2.0.php

**Comment by fabiangreffrath** [comment]
I'll check both restarting renderer and taking screenshots with this approach tomorrow, when I am working on a Windows system.

**Comment by fabiangreffrath** [comment]
I can confirm that with this approach both mode changing and screenshots work in Windows 11.

**Pull request merged**